samoa-api/src/main/java/org/apache/samoa/streams/kafka/KafkaReader.java

```diff
@@ -0,0 +1,279 @@
+                    .addFetch(a_topic, a_partition, readOffset, 100000)
+                    .build();
+            FetchResponse fetchResponse = null;
+
+            try {
+                fetchResponse = consumer.fetch(req);
+            } catch (Exception e) {
+
+            }
+
+            /**
+             * SimpleConsumer does not handle lead broker failures, you have to handle it
+             *  once the fetch returns an error, we log the reason, close the consumer then try to figure
+             *  out who the new leader is
+             */
+            if (fetchResponse.hasError()) {
+                numErrors++;
+                // Something went wrong!
+                short code = fetchResponse.errorCode(a_topic, a_partition);
+                System.out.println("Error fetching data from the Broker:" + leadBroker + " Reason: " + code);
+                if (numErrors > 5) break;
+                if (code == ErrorMapping.OffsetOutOfRangeCode()) {
+                    // We asked for an invalid offset. For simple case ask for the last element to reset
+                    //readOffset = getLastOffset(consumer,a_topic, a_partition, kafka.api.OffsetRequest.LatestTime(), clientName);
+                    continue;
+                }
+                consumer.close();
+                consumer = null;
+                try {
+                    leadBroker = findNewLeader(leadBroker, a_topic, a_partition, a_port);
+                } catch (Exception e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+                continue;
+            }
+            //End Error handling
+
+            // Reading data cont.
+            numErrors = 0;
+
+            long numRead = 0;
+            Iterator it = (Iterator) fetchResponse.messageSet(a_topic, a_partition).iterator();
+            MessageAndOffset messageAndOffset = null;
+
+            try {
+                messageAndOffset = (MessageAndOffset) it.next();
+            } catch (Exception e) {
+                return null;
+            }
+
+            //for (MessageAndOffset messageAndOffset : fetchResponse.messageSet(a_topic, a_partition)) {
+            long currentOffset = messageAndOffset.offset();
+            if (currentOffset < readOffset) {
+                System.out.println("Found an old offset: " + currentOffset + " Expecting: " + readOffset);
+                continue;
+            }
+            readOffset = messageAndOffset.nextOffset();
+            ByteBuffer payload = messageAndOffset.message().payload();
+
+            byte[] bytes = new byte[payload.limit()];
+            payload.get(bytes);
+            try {
+                System.out.println(String.valueOf(messageAndOffset.offset()) + ": " + new String(bytes, "UTF-8"));
+                answer = String.valueOf(messageAndOffset.offset()) + ": " + new String(bytes, "UTF-8");
+                returnInstances.add(answer);
```

Comment:
Data read from Kafka can be generic and later can be parsed to instances. Using string as a message from Kafka will provide easier options for parsing.